moved/src/state_actor.rs

```diff
@@ -225,11 +225,19 @@ impl<
 
     fn execute_transactions(
         &mut self,
-        transactions: impl Iterator<Item = (B256, NormalizedExtendedTxEnvelope, L1GasFeeInput)>,
+        mut transactions: impl Iterator<Item = (B256, NormalizedExtendedTxEnvelope, L1GasFeeInput)>,
         base_fee: U256,
     ) -> ExecutionOutcome {
         let mut total_tip = U256::ZERO;
         let mut outcomes = Vec::new();
+        let first_tx = transactions.next();
+        let l1_fee = match first_tx.as_ref() {
+            Some((_, NormalizedExtendedTxEnvelope::DepositedTx(deposit_tx), _)) => {
+                Some(self.l1_fee.for_deposit(deposit_tx.data.as_ref()))
+            }
+            _ => None,
+        };
+        let transactions = iter::once(first_tx).flatten().chain(transactions);
```

Comment:
I think a better way to do this is to make the original iterator [peekable](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.peekable) and [peek](https://doc.rust-lang.org/std/iter/struct.Peekable.html#method.peek) at the first element instead of popping it off and then making a new iterator from this chained with the tail.